src/solver/conv_direct_naive_conv_bwd.cpp

```diff
@@ -202,17 +223,37 @@ ConvSolution ConvDirectNaiveConvBwd::GetSolution(const ConvolutionContext& ctx,
                 }
             };
         };
+    }
     else
+    {
         result.invoker_factory = [=](const std::vector<Kernel>& kernels) {
             const auto kern = kernels[0];
             return [=](const Handle& handle, const AnyInvokeParams& primitive_parameters) {
                 decltype(auto) data_ctx = primitive_parameters.CastTo<conv::DataInvokeParams>();
                 const auto& tensors     = data_ctx.tensors;
                 float elapsed           = 0;
 
+                auto in_strides = MakeStrideArray<6>(
```

Comment:
@atamazov : Addressing all [Perf] comments related to `MakeStrideArray`. I don't think we can move any of these calls out of the `Invoker` lambda as they depend on the strides stored in tensor descriptors themselves.

Reply:
You can. Please read https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1958. Please do not hesitate to ask if something will remain unclear.

Reply:
@atamazov is there any guarantees that exactly the same tensors will be passed into invoker lambda?
I mean tensor strides is not a part of convolution parameters.
But it looks like we rely on this functionality in winograd convolution as well, so it seems strides and tensors should be the same for `GetSolution` and lambda invocation.

Reply:
@atamazov : we discussed this today and concluded that we cannot move stride computation out of `Invoker` lambda because someone may call the same convolution with `in,out,wei` tensors a second time such that the strides of these tensors are different from the tensors passed the first time. e.g, someone calls convolution with `in_1, wei, out_1` and then `in_2, wei, out_2` where strides are different between the two sets of inputs. 

So I propose keeping this computation in the `Invoker` lambda for now. We can think of ways to move this computation out later on by, for example, adding strides to the invoker lookup key string.

Reply:
@amberhassaan Unfortunately you misunderstand the concept of Invoker a bit. It's my fault probably.

We build dedicated invoker per each problem config. Before proceeding, please read discussion at https://github.com/ROCmSoftwarePlatform/MIOpen/pull/441#discussion_r504292283 (it is mentioned in ticket #216 "Design of Invokers").

If someone will call the "same" convolution but with different strides this is actually different problem config, i.e. different convolution. We library shall create dedicated Invoker for it.

BTW I forgot to remind you of some important changes that are needed. These will be described in a separate review comment.

Reply:
@CAHEK7 
> I mean tensor strides is not a part of convolution parameters.

Tensor descriptors (including their strides) are part of a convolution problem.

Reply:
If I understand correctly, we will take it up in the follow up work and leave things as is in the Invoker lambda for now? @atamazov .

Reply:
> f someone will call the "same" convolution but with different strides this is actually different problem config, i.e. different convolution. We library shall create dedicated Invoker for it.

UPDATE:

In discussion with @JehandadKhan, we decided not to include strides of non-packed tensors to the database keys. Only an optional flag (saying that at least one tensor is non-packed) should be included there. The above means that databases will share the same find-db records, same Invoker instances and same perf-db information for the non-packed convolutions that differ only in strides.

The above design should work correctly provided that:
- If an Invoker instance is able to compute some non-packed convolution, then the same instance must be able to compute any similar non-packed convolution that differs only in strides.
  - [Consequence 1] An Invoker that is used for computation of non-packed convolutions must read stride information from `DataInvokeParams` (passed to Invoker) and pass it to the kernels as run-time parameters. 
  - [Consequence 2] Stride information should not be used for building any compile-time parameters.
- If a kernel that is used to compute a non-packed convolution requires tuning, then the same tuning parameters must provide similar performance for non-packed convolutions that differs only in strides.

From now on, the above is becoming a part of the Solver/Solution/Invoker specification.

@amberhassaan @junliume

Reply:
[resolved] But please do not hide this comment.